### ISSUE TITLE
how to improve map-matching on sharp curves?

### ISSUE THREAD

**Issue by fog9001** [valhalla/valhalla#4773]
Hi,
Although the results of map matching with Valhalla are really good, I can't find a way to resolve some of the issues encountered, mainly in very sharp curves and roundabouts.

I tried different "gps_accuracy", "search_radius", and "costing" values, both with and without a timestamp, but there were no significant improvements.

![image](https://github.com/valhalla/valhalla/assets/1434368/41613a02-c0e8-4845-bd30-216ae02ba2ad)
![image](https://github.com/valhalla/valhalla/assets/1434368/38b440a4-f3c4-4b1f-b208-a08e3ccaddfa)

 The accuracy of the provided GPS coordinates is not very high, but I am surprised by the poor results obtained in these scenarios.

Do you have any suggestions or ideas I could try to improve the results?

**Comment by dnesbitt61** [comment]
I am starting to look into this as I am seeing similar results at intersections like the one below:
![image](https://github.com/valhalla/valhalla/assets/1838768/98432a4d-176d-44db-818a-a6bc9917f267)
Map matching fails here, adding a "break" (separate route).

**Comment by kevinkreiser** [comment]
yeah looks like a regression to me because it certainly used to work however i doubt its a recent regression as we havent made many changes to that code for a good 3 years. given that you see it at interesections we'll probably want to look at what happens when the matcher uses a node snap for edge candidates. i know there is special code to handle that puts in both incoming and outgoing edges from the node (ie edges with distance along set to both 100% and 0%)

**Comment by M393** [comment]
https://github.com/valhalla/valhalla/issues/3638#issuecomment-1131476431 is probably the same issue:

**Comment by dnesbitt61** [comment]
hmmm...I cannot seem to create a simple request that breaks (for the simple intersection case above). I was testing using traces from old routes and found this case as part of a longer route. I think I may have had a small search radius (15m). The geometry matched OSM road geometry at this intersection from 5 years ago. I was testing how map-matching quality can degrade over time due to OSM geometry changes. Anyway, ignore my comment above until/unless I can isolate an example.

**Comment by M393** [comment]
@dnesbitt61 There is an example request in https://github.com/valhalla/valhalla/issues/3638#issuecomment-1131476431

**Comment by henkjan68** [comment]
https://github.com/valhalla/valhalla/issues/3638#issuecomment-1131476431 gives a correct result when using the public valhalla service. Has this been fixed maybe?
